### PR TITLE
make: pkg.mk: don't call `git am` if there are no patches

### PR DESCRIPTION
**Pull request by kaspar030**
### Contribution description

Don't call `git am` if there are no patches. Git *should* gracefully handle it, but somehow doesn't always (like, /dev/null broken).
And, no need to call `git` if a simple test is sufficient.

### CODE REVIEW

File: pkg/pkg.mk [hunk 1]
```diff
@@ -115,7 +115,9 @@ $(PKG_PATCHED): $(PKG_PATCHED_PREREQUISITES)
 	$(call gen_dependency_files,$@.d,$(PKG_PATCHED_PREREQUISITES))
 	$(Q)$(GIT_IN_PKG) clean $(GIT_QUIET) -xdff '**' -e $(PKG_STATE:$(PKG_SOURCE_DIR)/%='%*')
 	$(Q)$(GIT_IN_PKG) checkout $(GIT_QUIET) -f $(PKG_VERSION)
-	$(Q)$(GIT_IN_PKG) $(GITFLAGS) am $(GITAMFLAGS) $(PKG_PATCHES) </dev/null
+	$(Q) if test -n "$(PKG_PATCHES)" ; then \
+	       $(GIT_IN_PKG) $(GITFLAGS) am $(GITAMFLAGS) $(PKG_PATCHES) </dev/null ; \
```
Comment (kfessel) [inline pkg/pkg.mk:119]:
```suggestion
	       $(GIT_IN_PKG) $(GITFLAGS) am $(GITAMFLAGS) $(PKG_PATCHES); \
```
no need for that '/dev/null'

Comment (kaspar030) [reply pkg/pkg.mk]: yup, squashed right in. but it wouldn't hurt. :/

**Comment by kfessel** [comment]
The reason for that `<dev/null` is a non graceful handling of no file given (`git am`  will try to read a patch from stdin)
The solution of this PR is check if git am should even be called (instead of piping `/dev/null` into stdin)

another possible source of nothing could be `echo -n |`

`
$(Q) echo -n | $(GIT_IN_PKG) $(GITFLAGS) am $(GITAMFLAGS) $(PKG_PATCHES)
`

**Comment by kaspar030** [comment]
> another possible source of nothing could be `echo -n |`

yeah, but that `-n` isn't posix?

even /dev/null was a fine solution. I had this fix done for the case where `/dev/null` was broken, before I understood that, to fix "empty patch" errors from git. I'd also be fine with just closing this. let's not spend more time here

**Comment by kfessel** [comment]
> yeah, but that `-n` isn't posix?

posix is seems to be aware of `-n` https://www.unix.com/man-page/posix/1posix/echo/ but recommends using `printf`
so  `printf "" |` would do

**Review by kfessel — APPROVED** [review]
murdock tested this  and it is working as intended

**Pull request merged**